bulldozer/merge_test.go

```diff
@@ -30,9 +35,20 @@ type MockMerger struct {
 
 	DeleteCount int
 	DeleteError error
+
+	GithubStatusCode int
+	GithubMessage    string
 }
 
 func (m *MockMerger) Merge(ctx context.Context, pullCtx pull.Context, method MergeMethod, msg CommitMessage) (string, error) {
+	if m.GithubStatusCode != 0 {
+		return "", github.CheckResponse(
+			&http.Response{
+				StatusCode: m.GithubStatusCode,
+				Body:       io.NopCloser(bytes.NewReader([]byte(fmt.Sprintf(`{"message": "%s"}`, m.GithubMessage)))),
+			},
+		)
```

Comment:
I think we could simplify this by using `github.CheckResponse` to set the the existing `MergeError` field as part of the test, instead of modifying the mock type to understand GitHub statuses.

Reply:
Thanks, that is a big improvement!